egui_glow/src/post_process.rs

```diff
@@ -0,0 +1,244 @@
+    ) -> Result<PostProcess, String> {
+        let fbo = unsafe { gl.create_framebuffer() }?;
+        unsafe {
+            gl.bind_framebuffer(glow::FRAMEBUFFER, Some(fbo));
+        }
+
+        let texture = unsafe { gl.create_texture() }.unwrap();
+        unsafe {
+            gl.bind_texture(glow::TEXTURE_2D, Some(texture));
+        }
+        unsafe {
+            gl.tex_parameter_i32(
+                glow::TEXTURE_2D,
+                glow::TEXTURE_WRAP_S,
+                glow::CLAMP_TO_EDGE as i32,
+            );
+        }
+        unsafe {
+            gl.tex_parameter_i32(
+                glow::TEXTURE_2D,
+                glow::TEXTURE_WRAP_T,
+                glow::CLAMP_TO_EDGE as i32,
+            );
+        }
+        unsafe {
+            gl.tex_parameter_i32(
+                glow::TEXTURE_2D,
+                glow::TEXTURE_MIN_FILTER,
+                glow::NEAREST as i32,
+            );
+        }
+        unsafe {
+            gl.tex_parameter_i32(
+                glow::TEXTURE_2D,
+                glow::TEXTURE_MAG_FILTER,
+                glow::NEAREST as i32,
+            );
+        }
+        unsafe {
+            gl.pixel_store_i32(glow::UNPACK_ALIGNMENT, 1);
+        }
+        let (internal_format, format) = if is_webgl_1 {
+            (glow::SRGB_ALPHA, glow::SRGB_ALPHA)
+        } else {
+            (glow::SRGB8_ALPHA8, glow::RGBA)
+        };
+
+        unsafe {
+            gl.tex_image_2d(
+                glow::TEXTURE_2D,
+                0,
+                internal_format as i32,
+                width,
+                height,
+                0,
+                format,
+                glow::UNSIGNED_BYTE,
+                None,
+            );
+            let error_code = gl.get_error();
+            assert_eq!(
+                error_code,
+                glow::NO_ERROR,
+                "Error occurred in post process texture initialization. code : 0x{:x}",
+                error_code
+            );
+        }
+        unsafe {
+            gl.framebuffer_texture_2d(
+                glow::FRAMEBUFFER,
+                glow::COLOR_ATTACHMENT0,
+                glow::TEXTURE_2D,
+                Some(texture),
+                0,
+            );
+        }
+
+        unsafe {
+            gl.bind_texture(glow::TEXTURE_2D, None);
+        }
+        unsafe {
+            gl.bind_framebuffer(glow::FRAMEBUFFER, None);
+        }
+
+        let vert_shader = compile_shader(
+            gl,
+            glow::VERTEX_SHADER,
+            include_str!("shader/post_vertex_100es.glsl"),
+        )?;
+        let frag_shader = compile_shader(
+            gl,
+            glow::FRAGMENT_SHADER,
+            include_str!("shader/post_fragment_100es.glsl"),
+        )?;
+        let program = link_program(gl, [vert_shader, frag_shader].iter())?;
+
+        let positions = vec![0.0f32, 0.0, 1.0, 0.0, 0.0, 1.0, 1.0, 1.0];
+
+        let indices = vec![0u8, 1, 2, 1, 2, 3];
+        unsafe {
```

Comment:
Instead of having so many this functions has so many `unsafe` blocks I think it just nicer to add one big around the whole function, or mark the entire function as `unsafe`